troubleshooting/debugging.md

```diff
@@ -1498,5 +1504,114 @@ If you use Amazon ECS FireLens, then standard out and standard error will be col
+
+Note: *If you have multiple outputs sending to multiple destinations, make sure you compute this statistic for all outputs. Remember to divide the total ingestion rate by the number of Fluent Bit instances to get the ingestion rate for a single Fluent Bit.*
+
+* **CloudWatch Logs**: If you use CloudWatch, use the [`IncomingBytes` metric in CloudWatch Metrics to track the ingestion rate to your log group(s)](https://repost.aws/knowledge-center/cloudwatch-logs-bill-increase). Assuming that all Fluent Bit instances send at roughly the same rate, you can then divide the log group ingestion rate by the number of Fluent Bit instances. Then you have the rate for each individual Fluent Bit. 
+* **Kinesis Firehose**: If you use Firehose, use the [`IncomingBytes` and `IncomingRecords` metrics in CloudWatch Metrics to track the ingestion rate to your stream(s)](https://docs.aws.amazon.com/firehose/latest/dev/monitoring-with-cloudwatch-metrics.html). Assuming that all Fluent Bit instances send at roughly the same rate, you can then divide the stream ingestion rate by the number of Fluent Bit instances. Then you have the rate for each individual Fluent Bit. 
+* **Kinesis Streams**: If you use Kinesis, use the [`IncomingBytes` metric in CloudWatch Metrics to track the ingestion rate to your streams(s)](https://docs.aws.amazon.com/streams/latest/dev/monitoring-with-cloudwatch.html). Assuming that all Fluent Bit instances send at roughly the same rate, you can then divide the stream ingestion rate by the number of Fluent Bit instances. Then you have the rate for each individual Fluent Bit. 
+* **S3**: If you use S3, one option is simply to search objects by timestamp and sum the sizes for some time period. Alternativel, you can use the [`BytesUploaded` metric in CloudWatch Metrics to track the ingestion rate to your bucket(s)](https://docs.aws.amazon.com/AmazonS3/latest/userguide/metrics-dimensions.html#s3-request-cloudwatch-metrics). Assuming that all Fluent Bit instances send at roughly the same rate, you can then divide the bucket ingestion rate by the number of Fluent Bit instances. Then you have the rate for each individual Fluent Bit. 
+
+An alternative that works for all outputs is to use the [Fluent Bit monitoring server](https://docs.fluentbit.io/manual/administration/monitoring). You can access it after SSH into the instance/node that Fluent Bit is running on (or use [ECS Exec on ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html), [kubectl exec on K8s](https://kubernetes.io/docs/tasks/debug/debug-application/get-shell-running-container/)). You must enable it in your configuration to use it. The monitoring server provides a `fluentbit_output_proc_bytes_total` counter metric for each output. Sum them together and divide by runtime to get the ingestion rate. 
+
+##### Recommendations
+
+If a single Fluent Bit instance will send (in total across all outputs) at a sustained rate of several MB/s or more, then we recommend:
```

Comment:
also add this as recommendation: https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/log-driver-buffer-limit